docs/source/topics/data_types.rst

```diff
@@ -0,0 +1,508 @@
+.. _topics:data_types:
+
+**********
+Data types
+**********
+
+AiiDA already ships with a number of useful data types.
+This section details the most common, and some handy features/functionalities to work with them.
+
+.. _topics:data_types:core:
+
+Core data types
+===============
+
+Here follows a short summary of common data types already provided with AiiDA.
+This list is not complete, see also inside :py:mod:`aiida.orm.nodes.data` for the list of all available plugins.
+
+We also mention, when relevant, what is stored in the database (as attributes, so that it can be easily queried e.g. with the :ref:`QueryBuilder <how-to:data:find>`) and what is stored in the file repository (providing access to the file contents, but not efficiently queryable: this is useful for e.g. big data files that don't need to be queried for).
+
+For all data types, you can follow the link to the data class to read more about the methods provided, how to access them, and so on.
```

Comment:
If this is in the "reference" section, should we say something like "follow the link to the reference page on the corresponding data class" or similar?